.github/workflows/ci_benchmark.yml

```diff
@@ -0,0 +1,150 @@
+            return repository.name === 'astropy' && repository.owner.login === 'astropy';
+
+  benchmark:
+    needs: check
+    if: ${{ needs.check.outputs.result == 'true' }}
+    name: "Compare asv with astropy main"
+    runs-on: ubuntu-latest
+    env:
+      CCACHE_BASEDIR: "${{ github.workspace }}"
+      CCACHE_DIR: "${{ github.workspace }}/.ccache"
+      CCACHE_COMPRESS: true
+      CCACHE_COMPRESSLEVEL: 6
+      CCACHE_MAXSIZE: 400M
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v5
+        name: Install Python
+        with:
+          python-version: "3.11"
+
+      - name: Setup some dependencies
+        shell: bash -l {0}
+        run: |
+          sudo apt-get update -y && sudo apt-get install -y ccache
+          # Make gcc/gxx symlinks first in path
+          sudo /usr/sbin/update-ccache-symlinks
+          echo "/usr/lib/ccache" >> $GITHUB_PATH
+
+      - name: "Prepare ccache"
+        id: prepare-ccache
+        shell: bash -l {0}
+        run: |
+          echo "key=benchmark-$RUNNER_OS" >> $GITHUB_OUTPUT
+          echo "timestamp=$(date +%Y%m%d-%H%M%S)" >> $GITHUB_OUTPUT
+          ccache -p
+          ccache -z
+
+      - name: "Restore ccache"
+        uses: actions/cache@v4
+        with:
+          path: .ccache
+          key: ccache-${{ secrets.CACHE_VERSION }}-${{ steps.prepare-ccache.outputs.key }}-${{ steps.prepare-ccache.outputs.timestamp }}
+          restore-keys: |
+            ccache-${{ secrets.CACHE_VERSION }}-${{ steps.prepare-ccache.outputs.key }}-
+
+      - name: Run benchmarks
+        shell: bash -l {0}
+        id: benchmark
+        env:
+          OPENBLAS_NUM_THREADS: 1
+          MKL_NUM_THREADS: 1
+          OMP_NUM_THREADS: 1
+          ASV_FACTOR: 1.3
+          ASV_SKIP_SLOW: 1
+        run: |
+          set -x
+          python -m pip install asv virtualenv packaging
+
+          git clone -b main https://github.com/astropy/astropy-benchmarks.git --single-branch
+
+          # ID this runner
+          python -m asv machine --yes --conf asv.ci.conf.json
+
+          echo "Baseline:  ${{ github.event.pull_request.base.sha }} (${{ github.event.pull_request.base.label }})"
+          echo "Contender: ${GITHUB_SHA} (${{ github.event.pull_request.head.label }})"
+
+          # Run benchmarks for current commit against base
+          ASV_OPTIONS="--split --show-stderr --factor $ASV_FACTOR --conf asv.ci.conf.json"
+          python -m asv continuous $ASV_OPTIONS ${{ github.event.pull_request.base.sha }} ${GITHUB_SHA} \
+              | sed "/Traceback \|failed$\|PERFORMANCE DECREASED/ s/^/::error::/" \
+              | tee benchmarks.log
+
+      - name: "Check ccache performance"
+        shell: bash -l {0}
+        run: ccache -s
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: asv-benchmark-results
+          path: benchmarks.log
+
+      - name: Download artifact
```

Comment:
I made 2 different workflows at first to make the commenting work but didn't remove this.
Should we just remove all the commenting bits from the PR? To enable comments we need to run with `pull_request_target` event but this event can't run code from the PR. It runs on the base commit of the PR.

Reply:
> Should we just remove all the commenting bits from the PR? 

If we are not using it, let's remove it for now. Otherwise, it would just rot. We can always reintroduce later if needed. Thanks!